pkg/unpack/reflector.go

```diff
@@ -222,6 +222,14 @@ func convertStruct(structPtr reflect.Value, in map[string]interface{}) error {
 		case reflect.Ptr:
 			derefType := emptyFieldVal.Type().Elem()
 			if derefType.Kind() == reflect.Struct {
+				subVal, ok := o.(reflect.Value)
+				if ok {
+					if subVal.Type().AssignableTo(emptyFieldVal.Type()) {
+						emptyFieldVal.Set(subVal)
+						continue
+					}
+					return fmt.Errorf("JSON field '%s': cannot assign value of type '%s' inside of struct type '%s'", fieldName, goName(subVal), goName(val))
```

Comment:
```suggestion
					return fmt.Errorf("JSON field %q: cannot assign value of type %q inside of struct type %q", fieldName, goName(subVal), goName(val))
```

Reply:
I will change this throughout.  This was written before  `%q` was in my repertoire :-)